geopandas/vectorized.pyx

```diff
@@ -1615,13 +1510,21 @@ cpdef cysjoin(np.ndarray[np.uintp_t, ndim=1, cast=True] left,
 
         cfree(sv.a)
 
-    finally:
-        GEOS_finish_r(handle)
-
     return left_out, right_out
 
 
 def concat(L):
     L = list(L)
     x = np.concatenate([ga.data for ga in L])
     return GeometryArray(x, base=set(L))
+
+
+cdef class get_handle:
+    cdef GEOSContextHandle_t handle
+
+    cdef GEOSContextHandle_t __enter__(self):
+        self.handle = GEOS_init_r()
+        return self.handle
+
+    def __exit__(self, a, b, c):
```

Comment:
a, b, c leftover from example? or are it required args

Reply:
Ah indeed, I've renamed these to their proper names